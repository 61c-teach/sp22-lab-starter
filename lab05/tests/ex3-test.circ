<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<project source="3.4.3-61c.cc0f4a6" version="1.0">
  This file is intended to be loaded by Logisim-evolution (https://github.com/reds-heig/logisim-evolution).

  <lib desc="#Wiring" name="0">
    <tool name="Splitter">
      <a name="appear" val="center"/>
      <a name="facing" val="west"/>
    </tool>
    <tool name="Pin">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Probe">
      <a name="radix" val="16"/>
    </tool>
    <tool name="Tunnel">
      <a name="width" val="2"/>
    </tool>
    <tool name="Pull Resistor">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Clock">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Constant">
      <a name="value" val="0xff"/>
      <a name="width" val="8"/>
    </tool>
  </lib>
  <lib desc="#Gates" name="1">
    <tool name="PLA">
      <a name="table" val=""/>
    </tool>
  </lib>
  <lib desc="#Plexers" name="2">
    <tool name="Multiplexer">
      <a name="enable" val="false"/>
    </tool>
    <tool name="Demultiplexer">
      <a name="enable" val="false"/>
    </tool>
  </lib>
  <lib desc="#Arithmetic" name="3"/>
  <lib desc="#Memory" name="4">
    <tool name="D Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="T Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="J-K Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="S-R Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Register">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Counter">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Shift Register">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Random">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="RAM">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="ROM">
      <a name="appearance" val="logisim_evolution"/>
      <a name="contents">addr/data: 8 8
0
</a>
    </tool>
  </lib>
  <lib desc="#Base" name="5">
    <tool name="Text Tool">
      <a name="font" val="SansSerif plain 12"/>
      <a name="halign" val="center"/>
      <a name="text" val=""/>
      <a name="valign" val="base"/>
    </tool>
  </lib>
  <lib desc="file#../ex3.circ" name="6"/>
  <main name="main"/>
  <options>
    <a name="gateUndefined" val="ignore"/>
    <a name="simlimit" val="1000"/>
    <a name="simrand" val="0"/>
  </options>
  <mappings>
    <tool lib="5" map="Button2" name="Menu Tool"/>
    <tool lib="5" map="Button3" name="Menu Tool"/>
    <tool lib="5" map="Ctrl Button1" name="Menu Tool"/>
  </mappings>
  <toolbar>
    <tool lib="5" name="Poke Tool"/>
    <tool lib="5" name="Edit Tool"/>
    <tool lib="5" name="Wiring Tool"/>
    <tool lib="5" name="Text Tool">
      <a name="font" val="SansSerif plain 12"/>
      <a name="halign" val="center"/>
      <a name="text" val=""/>
      <a name="valign" val="base"/>
    </tool>
    <sep/>
    <tool lib="0" name="Pin">
      <a name="appearance" val="NewPins"/>
    </tool>
    <tool lib="0" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="output" val="true"/>
    </tool>
    <sep/>
    <tool lib="1" name="NOT Gate"/>
    <tool lib="1" name="AND Gate"/>
    <tool lib="1" name="OR Gate"/>
    <tool lib="1" name="XOR Gate"/>
    <tool lib="1" name="NAND Gate"/>
    <tool lib="1" name="NOR Gate"/>
    <sep/>
    <tool lib="4" name="D Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool lib="4" name="Register">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
  </toolbar>
  <circuit name="main">
    <a name="appearance" val="logisim_evolution"/>
    <a name="circuit" val="main"/>
    <a name="circuitnamedboxfixedsize" val="true"/>
    <a name="circuitvhdlpath" val=""/>
    <a name="clabel" val=""/>
    <a name="clabelfont" val="SansSerif bold 16"/>
    <a name="clabelup" val="east"/>
    <comp lib="0" loc="(190,100)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="CYCLE"/>
      <a name="locked" val="false"/>
      <a name="output" val="true"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(190,50)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="halt"/>
      <a name="locked" val="false"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(490,310)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="ADD_OUT"/>
      <a name="locked" val="false"/>
      <a name="output" val="true"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(490,350)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="REG_OUT"/>
      <a name="locked" val="false"/>
      <a name="output" val="true"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(60,110)" name="Constant">
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(60,40)" name="Constant">
      <a name="value" val="0xf"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(90,140)" name="Clock"/>
    <comp lib="3" loc="(100,100)" name="Adder"/>
    <comp lib="3" loc="(100,50)" name="Comparator"/>
    <comp lib="4" loc="(110,70)" name="Register">
      <a name="appearance" val="logisim_evolution"/>
    </comp>
    <comp lib="6" loc="(400,320)" name="AddMachine"/>
    <wire from="(100,100)" to="(110,100)"/>
    <wire from="(100,140)" to="(100,320)"/>
    <wire from="(100,140)" to="(110,140)"/>
    <wire from="(100,320)" to="(180,320)"/>
    <wire from="(100,50)" to="(190,50)"/>
    <wire from="(170,100)" to="(180,100)"/>
    <wire from="(180,100)" to="(180,170)"/>
    <wire from="(180,100)" to="(190,100)"/>
    <wire from="(30,170)" to="(180,170)"/>
    <wire from="(30,60)" to="(30,90)"/>
    <wire from="(30,60)" to="(60,60)"/>
    <wire from="(30,90)" to="(30,170)"/>
    <wire from="(30,90)" to="(60,90)"/>
    <wire from="(400,320)" to="(480,320)"/>
    <wire from="(400,340)" to="(480,340)"/>
    <wire from="(480,310)" to="(480,320)"/>
    <wire from="(480,310)" to="(490,310)"/>
    <wire from="(480,340)" to="(480,350)"/>
    <wire from="(480,350)" to="(490,350)"/>
    <wire from="(90,140)" to="(100,140)"/>
  </circuit>
</project>
