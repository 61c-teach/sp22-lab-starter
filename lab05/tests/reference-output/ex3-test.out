CYCLE	ADD_OUT	REG_OUT
00000000	00000001	00000000
00000001	00000010	00000001
00000010	00000011	00000010
00000011	00000100	00000011
00000100	00000101	00000100
00000101	00000110	00000101
00000110	00000111	00000110
00000111	00001000	00000111
00001000	00001001	00001000
00001001	00001010	00001001
00001010	00001011	00001010
00001011	00001100	00001011
00001100	00001101	00001100
00001101	00001110	00001101
00001110	00001111	00001110
00001111	00010000	00001111
